grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -983,6 +984,9 @@ public boolean fixDuplicateResources(String index, Set<ConceptId> resourceVertex
                 otherResource.deleteNode();
             }
 
+            String newIndex = mainResource.getIndex();
+            mainResource.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
```

Comment:
Should this still have a comment indicating it's temporary? It's a lot nicer than the other one anyway.